File: Python/examples/get_object.py
# -*- coding=utf-8

from qcloud_cos import CosConfig
from qcloud_cos import CosS3Client
from qcloud_cos import CosServiceError
from qcloud_cos import CosClientError

secret_id = 'COS_SECRETID'     # 替换为用户的secret_id
secret_key = 'COS_SECRETKEY'     # 替换为用户的secret_key
region = 'COS_REGION'    # 替换为用户的region
token = None               # 使用临时密钥需要传入Token，默认为空,可不填
config = CosConfig(Region=region, SecretId=secret_id, SecretKey=secret_key, Token=token)  # 获取配置对象
client = CosS3Client(config)

# 下载对象
def get_object():
    #.cssg-snippet-body-start:[get-object]
    response = client.get_object(
        Bucket='examplebucket-1250000000',
        Key='exampleobject',
        Range='string',
        IfMatch='"9a4802d5c99dafe1c04da0a8e7e166bf"',
        IfModifiedSince='Wed, 28 Oct 2014 20:30:00 GMT',
        IfNoneMatch='"9a4802d5c99dafe1c04da0a8e7e166bf"',
        IfUnmodifiedSince='Wed, 28 Oct 2014 20:30:00 GMT',
        ResponseCacheControl='string',
        ResponseContentDisposition='string',
        ResponseContentEncoding='string',
        ResponseContentLanguage='string',
        ResponseContentType='string',
        ResponseExpires='string',
        VersionId='string'
    )
    
    #.cssg-snippet-body-end

# 下载对象
def get_object_comp():
    #.cssg-snippet-body-start:[get-object-comp]
    ####  获取文件到本地
    response = client.get_object(
        Bucket='examplebucket-1250000000',
        Key='picture.jpg',
    )
    response['Body'].get_stream_to_file('output.txt')
    
    #### 获取文件流
    response = client.get_object(
        Bucket='examplebucket-1250000000',
        Key='picture.jpg',
    )
    fp = response['Body'].get_raw_stream()
    print(fp.read(2))
    
    #### 设置 Response HTTP 头部
    response = client.get_object(
        Bucket='examplebucket-1250000000',
        Key='picture.jpg',
        ResponseContentType='text/html; charset=utf-8'
    )
    print(response['Content-Type'])
    fp = response['Body'].get_raw_stream()
    print(fp.read(2))
    
    #### 指定下载范围
    response = client.get_object(
        Bucket='examplebucket-1250000000',
        Key='picture.jpg',
        Range='bytes=0-10'
    )
    fp = response['Body'].get_raw_stream()
    print(fp.read())
    
    #.cssg-snippet-body-end

#.cssg-methods-pragma


# 下载对象
get_object()

# 下载对象
get_object_comp()

#.cssg-methods-pragma

File: php/examples/GetObject.php
<?php

use Qcloud\Cos\Client;
use Qcloud\Cos\Exception\ServiceResponseException;

class GetObject
{
    private $cosClient;

    private $uploadId;
    private $eTag;
    private $versionId;

    // 下载对象
    protected function getObject() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-object]
        try {
            $result = $cosClient->getObject(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Key' => 'exampleobject',
                'SaveAs' => 'path/to/localFile',
            )); 
            // 请求成功
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

    // 下载对象
    protected function getObjectRange() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-object-range]
        try {
            $result = $cosClient->getObject(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Key' => 'exampleobject',
                'Range' => 'bytes=0-10'
            )); 
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

    // 下载对象
    protected function getObjectWithVersionId() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-object-with-versionId]
        try {
            $result = $cosClient->getObject(array(
                'Bucket' => 'examplebucket-1250000000', //格式：BucketName-APPID
                'Key' => 'exampleobject',
                'VersionId' => 'exampleVersionId'
            )); 
            // 请求成功
            print_r($result);
        } catch (\Exception $e) {
            // 请求失败
            echo($e);
        }
        
        //.cssg-snippet-body-end
    }

    // 下载对象
    protected function getObjectComp() {
        $cosClient = $this->cosClient;
        //.cssg-snippet-body-start:[get-object-comp]
        # 下载文件
        ## getObject(下载文件)
        ### 下载到内存
        try {
            $bucket = "examplebucket-1250000000"; //存储桶，格式：BucketName-APPID
            $key = "exampleobject"; //对象在存储桶中的位置，即称对象键
            $result = $cosClient->getObject(array(
                'Bucket' => $bucket,
                'Key' => $key));
            // 请求成功
            echo($result['Body']);
        } catch (\Exception $e) {
            // 请求失败
            echo "$e\n";
        }
        
        ### 下载到本地
        try {
            $bucket = "examplebucket-1250000000"; //存储桶，格式：BucketName-APPID
            $key = "exampleobject"; //对象在存储桶中的位置，即称对象键
            $localPath = @"path/to/localFile";//下载到本地指定路径
            $result = $cosClient->getObject(array(
                'Bucket' => $bucket,
                'Key' => $key,
                'SaveAs' => $localPath));
        } catch (\Exception $e) {
            // 请求失败
            echo "$e\n";
        }
        
        ### 指定下载范围
        /*
         * Range 字段格式为 'bytes=a-b'
         */
        try {
            $bucket = "examplebucket-1250000000"; //存储桶，格式：BucketName-APPID
            $key = "exampleobject"; //对象在存储桶中的位置，即称对象键
            $localPath = @"path/to/localFile";//下载到本地指定路径
            $result = $cosClient->getObject(array(
                'Bucket' => $bucket,
                'Key' => $key,
                'Range' => 'bytes=0-10',
                'SaveAs' => $localPath));
        } catch (\Exception $e) {
            // 请求失败
            echo "$e\n";
        }
        
        ## getObjectUrl(获取文件 UrL)
        try {    
            $bucket = "examplebucket-1250000000"; //存储桶，格式：BucketName-APPID
            $key = "exampleobject"; //对象在存储桶中的位置，即称对象键
            $signedUrl = $cosClient->getObjectUrl($bucket, $key, '+10 minutes');
            // 请求成功
            echo $signedUrl;
        } catch (\Exception $e) {
            // 请求失败
            print_r($e);
        }
        
        //.cssg-snippet-body-end
    }

	//.cssg-methods-pragma

    protected function init() {
        $secretId = "COS_SECRETID"; //"云 API 密钥 SecretId";
        $secretKey = "COS_SECRETKEY"; //"云 API 密钥 SecretKey";
        $region = "COS_REGION"; //设置一个默认的存储桶地域
        $this->cosClient = new Qcloud\Cos\Client(
            array(
                'region' => $region,
                'schema' => 'https', //协议头部，默认为http
                'credentials'=> array(
                    'secretId'  => $secretId ,
                    'secretKey' => $secretKey)));
    }

    public function mGetObject() {
        $this->init();

        // 下载对象
        $this->getObject();

        // 下载对象
        $this->getObjectRange();

        // 下载对象
        $this->getObjectWithVersionId();

        // 下载对象
        $this->getObjectComp();

	    //.cssg-methods-pragma
    }
}
?>